cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -321,40 +343,64 @@ static void batched_compress_async(compression_type compression,
   CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess, "Error in compression");
 }
 
-bool is_aligned(void const* ptr, std::uintptr_t alignment) noexcept
+constexpr bool is_aligned(void const* ptr, std::uintptr_t alignment) noexcept
 {
   return (reinterpret_cast<std::uintptr_t>(ptr) % alignment) == 0;
 }
 
+// since some compressors have extreme temp memory requirements, scale back the number
+// of chunks to process until the request for temp memory succeeds.
+std::pair<rmm::device_buffer, size_t> compress_temp_buffer(compression_type compression,
+                                                           size_t num_chunks,
+                                                           size_t max_uncomp_chunk_size,
+                                                           rmm::cuda_stream_view stream)
+{
+  // give up after 4 attempts
+  for (int i = 0; i < 4; i++) {
```

Comment:
Should we give up here (outside of failing to allocate temp buffer for a single chunk)?

Reply:
Not sure what to do.  Originally I just had a `while (num_chunks > 1)`, and could switch back. I just didn't want to thrash *too* much, but then the allocate should fail pretty fast.

Reply:
The way I see it: If we give up early the `write_parquet` call will throw. If we keep trying smaller batches, it might succeed, albeit with pretty bad performance. The second option seems preferable to me in most cases.

Reply:
It's back to a while loop now

Reply:
Easy enough to break out of the loop at a given scale so performance doesn't tank.  Suggestions for a sensible number of tries welcome :)